request.go

```diff
@@ -73,6 +81,12 @@ func (client *Client) doJsonRequest(method, api string,
 	}
 	defer resp.Body.Close()
 
+	if err := client.getRateLimit(resp); err != nil {
+		if err.(*RateLimit).Remaining == 0 {
```

Comment:
yea I wasnt sure about that. it seems like something the author should be aware of then do there own retry vs the library doing it for them. Mostly since this can be a large amount of time since once you are out you could wait as long as a hour before you can retry again.